code/modules/mob/living/simple_animal/bot/vibebot.dm

```diff
@@ -7,46 +7,49 @@
 	anchored = FALSE
 	health = 25
 	maxHealth = 25
-	damage_coeff = list(BRUTE = 1, BURN = 1, TOX = 0, CLONE = 0, STAMINA = 0, OXY = 0)
 	pass_flags = PASSMOB | PASSFLAPS
 	light_system = MOVABLE_LIGHT
 	light_range = 7
 	light_power = 3
 
 	hackables = "vibing scanners"
 	bot_mode_flags = ~BOT_MODE_PAI_CONTROLLABLE
-	radio_key = /obj/item/encryptionkey/headset_service //doesn't have security key
-	radio_channel = RADIO_CHANNEL_SERVICE //Doesn't even use the radio anyway.
+	radio_key = /obj/item/encryptionkey/headset_service
+	radio_channel = RADIO_CHANNEL_SERVICE
 	bot_type = VIBE_BOT
-	data_hud_type = DATA_HUD_DIAGNOSTIC_BASIC // show jobs
+	data_hud_type = DATA_HUD_DIAGNOSTIC_BASIC
 	path_image_color = "#2cac12"
 
+	///The vibe ability given to vibebots, so sentient ones can still change their color.
+	var/datum/action/innate/vibebot_vibe/vibe_ability
 
 /mob/living/simple_animal/bot/vibebot/Initialize(mapload)
 	. = ..()
 	update_appearance()
 
-/mob/living/simple_animal/bot/vibebot/turn_off()
-	. = ..()
-	remove_atom_colour(TEMPORARY_COLOUR_PRIORITY)
-	update_appearance()
-
-/mob/living/simple_animal/bot/vibebot/proc/Vibe()
-	remove_atom_colour(TEMPORARY_COLOUR_PRIORITY)
-	add_atom_colour("#[random_color()]", TEMPORARY_COLOUR_PRIORITY)
-	set_light_color(color)
-	update_appearance()
+/mob/living/simple_animal/bot/vibebot/Destroy()
+	QDEL_NULL(vibe_ability)
+	return ..()
 
-/mob/living/simple_animal/bot/vibebot/proc/retaliate(mob/living/carbon/human/H)
+/mob/living/simple_animal/bot/vibebot/Login()
+	. = ..()
+	if(!. || !client)
+		return FALSE
+	vibe_ability = new()
+	vibe_ability.Grant(src)
+	return TRUE
 
+/mob/living/simple_animal/bot/vibebot/Logout()
+	QDEL_NULL(vibe_ability)
+	return ..()
```

Comment:
Why do we create and destroy this every time a login logout occurs?

AI vibe-bots can't use the action anyways so is there harm in keeping it around for if a client re-enters the bot?

Reply:
done but also did a few more changes if you can re-review (im sorry)